customer-support-endpoint/src/main/kotlin/org/ostelco/prime/support/resources/HoustonResources.kt

```diff
@@ -464,3 +468,59 @@ class CustomerResource {
         }
     }
 }
+
+/**
+ * Resource used to provision new sim profiles.
+ */
+@Path("/support/simprofile")
+
+class SimProfilesResource {
+    private val logger by getLogger()
+    private val storage by lazy { getResource<AdminDataSource>() }
+
+    @EnableTracing
+    @POST
+    @Path("{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun provisionSimProfile(@Auth token: AccessTokenPrincipal?,
+                            @NotNull
+                            @PathParam("id")
+                            id: String,
+                            @QueryParam("regionCode")
+                            regionCode: String,
+                            @QueryParam("profileType")
+                            profileType: String,
+                            @QueryParam("alias")
+                            alias: String): Response =
+            if (token == null) {
+                Response.status(Response.Status.UNAUTHORIZED)
```

Comment:
I personally would have prefered a  return Response.status .... .build() .  in each of the branches instead of this implicit generation of a builder that is then built at the very end. I think it'a harder to read..

Also, if the direct return was used, the whole else branch could be dropped, decreasing the level of indentation, which in general increases readability, although in this simple case I'm not certain it would have made much difference.

Reply:
Personally I prefer a `build()` at the end and only one `return`. Make the code easier to read.